### PR TITLE
GameAdminRanks and GameRound API endpoints

### PR DESCRIPTION
**Pull request by TheTyrant25**
[INTERNAL] [TOOLING]
## About the PR
Adds list, add, update and delete routes for `GameAdminRanks`, and add, update, and end routes for `GameRounds`.

### CODE REVIEW

File: goonstation.dme [hunk 1]
```diff
@@ -1180,6 +1185,9 @@ var/datum/preMapLoad/preMapLoad = new
 #include "code\modules\goonhub\api\routes\players\notes\DeletePlayerNotes.dm"
 #include "code\modules\goonhub\api\routes\players\notes\GetPlayerNotes.dm"
 #include "code\modules\goonhub\api\routes\players\notes\UpdatePlayerNotes.dm"
+#include "code\modules\goonhub\api\routes\rounds\AddGameRound.dm"
+#include "code\modules\goonhub\api\routes\rounds\EndGameRound.dm"
+#include "code\modules\goonhub\api\routes\rounds\UpdateGameRound.dm"
```
Comment (ZeWaka) [inline goonstation.dme:1190]: bro please stop putting files in a different path than the actual api puts them in, this should be `gameround`

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/goonhub/api/routes/rounds/UpdateGameRound.dm [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+/// PUT /rounds/{gameRound}
+/// Update a game round. This should be used when game round data we care about is set after the start of the round.
+/datum/apiRoute/gameround/update
```
Comment (ZeWaka) [inline code/modules/goonhub/api/routes/rounds/UpdateGameRound.dm:4]: this should be `rounds`..... it needs to reflect the api!

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/goonhub/api/routes/rounds/UpdateGameRound.dm (hunk 2 above)
Comment (TheTyrant25) [reply code/modules/goonhub/api/routes/rounds/UpdateGameRound.dm]: ah I have not been doing that as a hard rule, whoops. Will update all prs to match this

File: code/modules/goonhub/api/routes/rounds/EndGameRound.dm [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+/// PUT /rounds/end/{gameRound}
+/// End a game round.
+/datum/apiRoute/rounds/end
+	method = RUSTG_HTTP_METHOD_PUT
+	path = "/rounds/end"
+	routeParams = list("gameRound")
+	body = list("crashed")
```
Comment (ZeWaka) [inline code/modules/goonhub/api/routes/rounds/EndGameRound.dm:8]: no, this isn't just a string body. it's a full json with a type, do this properly.

File: code/modules/goonhub/api/routes/rounds/UpdateGameRound.dm [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+/// PUT /rounds/{gameRound}
+/// Update a game round. This should be used when game round data we care about is set after the start of the round.
+/datum/apiRoute/rounds/update
+	method = RUSTG_HTTP_METHOD_PUT
+	path = "/rounds"
+	routeParams = list("gameRound")
+	body = list("game_type")
```
Comment (ZeWaka) [inline code/modules/goonhub/api/routes/rounds/UpdateGameRound.dm:8]: no, this isn't just a string body. it's a full json with a type, do this properly.

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/goonhub/api/requests/gameround/EndGameRoundBody.dm [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+/datum/apiBody/rounds/end/VerifyIntegrity()
+	if (
+		isnull(src.crashed)
+	)
+		return FALSE
+
+/datum/apiBody/rounds/end/toJson()
+	return json_encode(list(
+		"crashed"				= src.crashed,
```
Comment (ZeWaka) [inline code/modules/goonhub/api/requests/gameround/EndGameRoundBody.dm:18]:
```suggestion
		"crashed"	= src.crashed,
```

File: code/modules/goonhub/api/requests/gameround/UpdateGameRoundBody.dm [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+/datum/apiBody/rounds/update/put/VerifyIntegrity()
+	if (
+		isnull(src.game_type)
+	)
+		return FALSE
+
+/datum/apiBody/rounds/update/put/toJson()
+	return json_encode(list(
+		"game_type"				= src.game_type,
```
Comment (ZeWaka) [inline code/modules/goonhub/api/requests/gameround/UpdateGameRoundBody.dm:18]:
```suggestion
		"game_type"	= src.game_type,
```

File: code/modules/goonhub/api/requests/rounds/AddGameRoundBody.dm [hunk 7]
```diff
@@ -0,0 +1,30 @@
+
+/datum/apiBody/rounds/post
+	var/map					= "string"
+	var/server_id			= "string"
+	var/rp_mode				= FALSE
```
Comment (ZeWaka) [inline code/modules/goonhub/api/requests/rounds/AddGameRoundBody.dm:5]:
```suggestion
	var/map			= "string"
	var/server_id	= "string"
	var/rp_mode		= FALSE
```

File: code/modules/goonhub/api/requests/rounds/AddGameRoundBody.dm [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+		|| isnull(src.server_id) \
+		|| isnull(src.rp_mode) \
+	)
+		return FALSE
+
+/datum/apiBody/rounds/post/toJson()
+	return json_encode(list(
+		"map"				= src.map,
+		"server_id"			= src.server_id,
+		"rp_mode"			= src.rp_mode
```
Comment (ZeWaka) [inline code/modules/goonhub/api/requests/rounds/AddGameRoundBody.dm:29]:
```suggestion
		"map"		= src.map,
		"server_id"	= src.server_id,
		"rp_mode"	= src.rp_mode
```

**Pull request merged**